docs/content/NewButton.mdx

```diff
@@ -0,0 +1,169 @@
+storybook: '/react/storybook?path=/story/composite-components-new-button--default-button'
+description: Button is core design component used for actions.
+---
+
+import {Props} from '../src/props'
+import {ComponentChecklist} from '../src/component-checklist'
+import {NewButton} from '@primer/components'
+
+## Usage
+
+### Installation
+
+```js
+import {NewButton} from '@primer/components/drafts'
+```
+
+### Default button
+
+Using the component without any explicit props gives you the default button.
+
+```jsx live
+<NewButton>Default</NewButton>
+```
+
+### Danger button
+
+This button is used to warn users about potentially destructive actions
+
+```jsx live
+<NewButton variant="danger">Danger</NewButton>
+```
+
+### Outline button
+
+This button is typically used as a secondary button
+
+```jsx live
+<NewButton variant="outline">Outline</NewButton>
+```
+
+### Invisible button
+
+This button indicates that the action is a low priority one.
+
+```jsx live
+<NewButton variant="invisible">Invisible</NewButton>
+```
+
+### Different sized buttons
+
+Buttons can be in three different sizes. `small`, `medium`, `large`.
+
+```jsx live
+<>
+  <NewButton size="small">Search</NewButton>
+  <NewButton sx={{mt: 2}}>Search</NewButton>
+  <NewButton sx={{mt: 2}} size="large">
+    Search
+  </NewButton>
+</>
+```
+
+### Icon and button
+
+We can place an icon inside the button in either the leading or the trailing position
+
+```jsx live
+<>
+  <NewButton leadingIcon={SearchIcon}>Search</NewButton>
+  <NewButton trailingIcon={SearchIcon} sx={{mt: 2}}>
+    Search
+  </NewButton>
+</>
+```
+
+### Icon only button
+
+This uses the `icon` prop and creates a nice square button with only the icon visible.
+
+```jsx live
+<NewButton icon={SearchIcon}>Search</NewButton>
+```
+
+### Different sized icon buttons
+
+Buttons can be in three different sizes. `small`, `medium`, `large`.
+
+```jsx live
+<>
+  <NewButton icon={SearchIcon} size="small">
+    Search
+  </NewButton>
+  <NewButton icon={SearchIcon} sx={{mt: 2}}>
+    Search
+  </NewButton>
+  <NewButton icon={SearchIcon} sx={{mt: 2}} size="large">
+    Search
+  </NewButton>
+</>
+```
+
+### Counter component
+
+A common use case for primer is a button with a counter component which shows count value as provided by the consumer.
+We provide `Button.Counter` as a composite component which requires you to provide a number as child.
+It renders a counter with the same `variant` styles as the parent Button component
+
+```jsx live
+<NewButton>
+  Watch
+  <NewButton.Counter>1</NewButton.Counter>
+</NewButton>
+```
+
+### Styling a button
+
+A button can be styled in an appropriate manner using the `sx` prop. This maybe to change width, or to add margins etc.
+Here's an example of a block button which takes 100% of available width. Checkout [styled-system](https://styled-system.com/) to see what you can send in an `sx` prop.
+
+```jsx live
+<NewButton sx={{width: '100%'}}>Block</NewButton>
```

Comment:
Question: 100% width is presumably a common use case for many Button users, so should we instead turn that into a `block` prop? `sx` example could then be for custom margins, as you described?

Reply:
Yeah, this is what I had initially implemented. Then I felt that the `Button` api would unnecessarily bloat with a new prop. It's just a small style addition, we can document it as a recipe.

Yesterday though, I found a bug. Basically if an icon and text need to be shown in a block width button, they show up with a large gap between them. I'm thinking I should revisit this block type of button in the near future. I'll put up an issue on this ASAP.